libs/api.lua

```diff
@@ -130,6 +132,45 @@ end
 
 local metaCache = {}
 
+-- Collect statistics about this server's running instance.
+-- Return these metrics in a table.  The intended consumer of this
+-- information should render it in JSON format for transmission to
+-- a time-series database and/or visualization tools.
+local function collectStats()
+  local entry, fdsUsed = 0, 0
+  local memoryUsed = 1024 * collectgarbage("count")
+
+  local function countFDs(path)
+    local entries = 0
+
+    -- You might think this leaks a filehandle, but it actually doesn't.
+    -- You can confirm this yourself by hitting /metrics several times in a row,
+    -- and observing that lua.fds.total does not increase with each GET request.
+
+    local iter = fs.scandir(path)
```

Comment:
Just a note, that is path doesn exist (ENOENT), then iter will be nil.  You can wrap fs.scandir in an assert if you want it to just throw in this case.